services/ui-src/src/utils/other/parsing.ts

```diff
@@ -51,3 +50,128 @@ export const sanitizeAndParseHtml = (html: string) => {
+      labelObject.hint && !labelObject.hideHint
+        ? `<p>${labelObject.hint}</p>`
+        : ""
+    }`;
+    return {
+      indicator,
+      label,
+    };
+  }
+  return {
+    indicator: "",
+    label: "",
+  };
+};
+
+const noResponse = `<p style="color:#9F142B">Not Answered</p>`;
+
+const returnChoices = (choiceItem: any, fieldData: any) => {
+  // loop through all choices
+  const dataItems: any[] = choiceItem.fieldData.map(
+    ({ value, key }: { value: string; key: string }) => {
+      const childItems: any[] = [];
+      // get ID of each choice
+      const choiceId = key.split("-")[1];
+      // check if choice has children
+      const children = choiceItem.parent.props.choices.find(
+        (choice: any) => choice.id === choiceId
+      ).children;
+
+      if (children) {
+        // loop through all children
+        children.map((childField: any) => {
+          if (typeof fieldData[childField.id] !== "string") {
+            if (childField.props) {
+              childItems.push(
+                `${
+                  parseFieldLabel({ ...childField.props, hideHint: true }).label
+                }${returnChoices(
+                  {
+                    parent: childField,
+                    fieldData: fieldData[childField.id],
+                  },
+                  fieldData
+                )}`
+              );
+              return;
+            }
+          }
+
+          childItems.push(
+            `${
+              childField.props
+                ? parseFieldLabel({ ...childField.props, hideHint: true }).label
+                : ""
+            }${parseAllLevels({ ...childField, fieldData })}`
+          );
+          return;
+        });
+      }
+
+      return `<p>${value}</p>${childItems.join(" ") || ""}`;
+    }
+  );
+
+  // Default return
+  return dataItems?.join(" ") || noResponse;
+};
+
+export const parseAllLevels = ({
+  fieldData,
+  id,
+  props,
+  type,
+  validation,
+}: any) => {
+  if (fieldData && fieldData[id]) {
+    const qualifier = props
+      ? props.mask || validation || type
+      : validation || type;
+
+    switch (qualifier) {
+      case "percentage":
+        return `${fieldData[id]}%`;
+      case "currency":
+        return `$${fieldData[id]}`;
+      case "email":
+      case "emailOptional":
+        return `<a href="mailto:${fieldData[id]}">${fieldData[id]}</a>`;
+      case "url":
+        return `<a href="${fieldData[id]}">${fieldData[id]}</a>`;
+      case "checkbox":
+      case "radio":
+        return returnChoices(
+          {
+            parent: { id, props, type, validation },
+            fieldData: fieldData[id],
+          },
+          fieldData
+        );
+      default:
+        return fieldData[id];
+    }
+  }
+
+  return noResponse;
+};
+
+export const parseDynamicFieldData = (data: any) => {
+  const formattedValues = data
+    ?.map((value: any) => `<p>${value?.name}</p>`)
+    .join(" ");
+  return formattedValues ?? noResponse;
+};
```

Comment:
Post-merge note: Stronger types

Reply:
Could maybe be combined with the formatter above and changed to use React.createElement.